generate-pdf.js

```diff
@@ -0,0 +1,97 @@
+const puppeteer = require("puppeteer");
+const yargs = require("yargs");
+
+DEFAULT_PRINT_OPTIONS = {
+  format: "Letter",
+  margin: {
+    top: "0.5in",
+    bottom: "0.5in",
+    right: "0.5in",
+    left: "0.5in"
+  }
+}
+
+GOTO_PDF_OPTIONS = {
+  waitUntil: "networkidle2"
+}
+
+VALIDATION_ERRORS = {
+  mustHaveWritepathOrRaw:
+    "At least one of the parameters { '--writePath <path>', '--raw' } must be supplied",
+  errorParsingJson:
+    "Error parsing option overrides as JSON: ",
+  urlIsMissing:
+    "The parameter '--url <url>' must be provided"
+}
+
+async function getUrlPdf(browser, url, writePath, optionOverrides) {
+  let pdfOptions = optionOverrides || DEFAULT_PRINT_OPTIONS;
+
+  pdfOptions.path = writePath;
+
+  const page = await browser.newPage();
+  await page.goto(url, GOTO_PDF_OPTIONS)
+  return await page.pdf(pdfOptions)
+}
+
+function validatePdfCliArgs(argv) {
```

Comment:
Some of the validation here seems redundant with the validation features yargs already has, like marking options as required. Can we use the built in features where possible?

Reply:
. I explicitly removed using yarg's required parameters feature in favor of what we have here. I had thought about it, and I have some reasons why I believe it is better as written

Advantages of what is here:
* All input errors are reported to the user, no matter what combination of input errors are supplued. Using demandOption, if an option listed in demandOptions is missing yargs will fail with an error demanding the option be supplied without calling the check method and will thus not report on other input errors.
* When reading the script, there is one and only one function/section of code to read which does all validation. I think it makes it easier for someone reading the script to not miss what validation is being done.

Reply:
Okay, sounds good.